metadata/metadata.c

```diff
@@ -211,13 +283,13 @@ PhysicalDiskRange *_get_physical_disk_range(Metadata *metadata, DiskRangeKey *ke
     return (PhysicalDiskRange *) bsearch(key_range, metadata->physical_disk_ranges, metadata->n_physical_disk_ranges, sizeof(PhysicalDiskRange *), _physical_disk_range_key_compare);
 }
 
-void _add_virtual_disk(Metadata *metadata, VirtualDisk *disk)
+void _add_virtual_disk(HyperwarpMetadata *metadata, VirtualDisk *disk)
 {
-    size_t last = metadata->n_virtual_disks;
-    metadata->virtual_disks = realloc(metadata->virtual_disks,
-                                      sizeof(VirtualDisk) * (last + 1));
-    metadata->virtual_disks[last] = disk;
-    metadata->n_virtual_disks = last + 1;
+    size_t last = metadata->metadata->n_virtual_disks;
+    metadata->metadata->virtual_disks = realloc(metadata->metadata->virtual_disks,
+                                        sizeof(VirtualDisk) * (last + 1));
+    metadata->metadata->virtual_disks[last] = disk;
+    metadata->metadata->n_virtual_disks = last + 1;
```

Comment:
this change doesn't seem necessary right now. You could just continue to pass the proto Metadata struct and it would be fine.